jax/experimental/unroll.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''Extensions to lax loop-related functions that partially unroll loops.'''
+
+from jax import api, lax
+
+from functools import partial
+from typing import Callable
+
+
+def unrolled_scan(f: Callable, init, xs, length=None, reverse=False):
+  with api.disable_jit():
```

Comment:
This seems like the wrong mechanism to achieve unrolling? It may work, but it is surprising and seems like it might have collateral damage.

Also, perhaps we should make `disable_jit` inside of a `jit` an error after we land omnistaging?